disboard/disboard/handlers/server/server.py

```diff
@@ -68,10 +81,32 @@ def get_status(self):
             status_time=dt.datetime.now(dt.timezone.utc) - provision_status.time if provision_status.time else None
         )
     
+    async def _poll_for_status(self, expected_status, handler):
+        logger.info("polling server %s to reach terminal state", self)
+        try:
+            while True:
+                await asyncio.sleep(IOTA)
+                instance_view = self._manager.get_vm_instance_view()
+                statuses = instance_view.statuses
+                if len(statuses) < 2:
+                    logging.info("no state for server %s, repolling", self)
+                    continue
+                current_status = statuses[1].code
+                logger.info("current state for server %s is %s, wanted %s", self, current_status, expected_status)
+                if current_status == expected_status:
+                    return await handler(True)
+                elif current_status in TERMINAL_STATUSES:
+                    return await handler(False)
+        except:
+            logger.error("unexpected error when polling for status of server %s: %s", self, traceback.format_exc())
```

Comment:
`logger.exception` will log with error severity and will automatically get the traceback. No need to import traceback either if u do this.